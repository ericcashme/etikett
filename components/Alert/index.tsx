import React, { FC, memo, useEffect, useState, useCallback } from 'react'
import { Button } from '../Button'
import * as S from './styles'
import { AlertProps } from './types'

export const Alert: FC<AlertProps> = memo(
  ({
    size = 'medium',
    close,
    action,
    children,
    isLoading = false,
    isDisabled = false,
    cancelButtonName = 'Cancel',
    confirmButtonName = 'Confirm'
  }) => {
    const [body] = useState(document?.querySelector('body') as HTMLBodyElement)

    const handleKeyUp = useCallback(
      (e) => {
        if (e.keyCode === 27) {
          e.preventDefault()
          close()
          window.removeEventListener('keyup', handleKeyUp, false)
        }
      },
      [close]
    )

    useEffect(() => {
      window.addEventListener('keyup', handleKeyUp, false)
      body.style.overflow = 'hidden'

      return () => {
        window.removeEventListener('keyup', handleKeyUp, false)
        body.style.overflow = 'visible'
      }
    }, [handleKeyUp, body])

    return (
      <S.Background>
        <S.Content size={size}>
          <S.InnerContent>{children}</S.InnerContent>

          <S.Actions>
            <Button kind="weak" isOutlined onClick={close}>
              {confirmButtonName}
            </Button>

            <Button kind="warning" isLoading={isLoading} isDisabled={isDisabled || isLoading} onClick={action}>
              {cancelButtonName}
            </Button>
          </S.Actions>
        </S.Content>
      </S.Background>
    )
  }
)

Alert.displayName = 'Alert'
