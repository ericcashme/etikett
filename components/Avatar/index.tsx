import React, { FC, memo } from 'react'
import Image from 'next/image'
import * as S from './styles'
import { AvatarProps } from './types'

export const Avatar: FC<AvatarProps> = memo(({ size = 'md', image, alt, hasBorder = false, ...rest }) => (
  <S.Wrap size={size} hasBorder={hasBorder} {...rest}>
    <Image src={image} alt={alt} width={S.getSize[size]} height={S.getSize[size]} layout="responsive" />
  </S.Wrap>
))

Avatar.displayName = 'Avatar'
