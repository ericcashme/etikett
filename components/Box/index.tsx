import React, { FC, memo } from 'react'
import * as S from './styles'
import { BoxProps } from './types'

export const Box: FC<BoxProps> = memo(
  ({
    type = 'default',
    kind = 'default',
    noRadius = false,
    backgroundUrl = '',
    backgroundOpacity = false,
    shadow = false,
    disabled = false,
    children,
    ...rest
  }) => (
    <S.Box
      type={type}
      kind={kind}
      noRadius={noRadius}
      backgroundUrl={backgroundUrl}
      backgroundOpacity={backgroundOpacity}
      shadow={shadow}
      disabled={disabled}
      {...rest}
    >
      {children}
    </S.Box>
  )
)

Box.displayName = 'Box'
