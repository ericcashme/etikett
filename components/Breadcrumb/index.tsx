import React, { FC, memo } from 'react'
import Link from 'next/link'
import { Link as LinkComp } from '../Link'
import * as S from './styles'
import { BreadcrumbProps, Page } from './types'

export const Breadcrumb: FC<BreadcrumbProps> = memo(({ pages, inverse = false }) => {
  const mutPages: Page[] = [...pages]
  const actualPage = mutPages.pop() as Page

  return (
    <S.Breadcrumb itemScope itemType="https://schema.org/BreadcrumbList">
      {mutPages.map((item, index) => (
        <S.BreadcrumbItem
          inverse={inverse}
          key={`bc-${item.path}`}
          itemProp="itemListElement"
          itemScope
          itemType="https://schema.org/ListItem"
        >
          <Link href={item.path} as={item.path} passHref>
            <LinkComp>
              <span itemProp="name">{item.label}</span>
              <meta itemProp="position" content={String(index + 1)} />
            </LinkComp>
          </Link>
        </S.BreadcrumbItem>
      ))}

      <S.BreadcrumbItem inverse={inverse} itemProp="itemListElement" itemScope itemType="https://schema.org/ListItem">
        <span itemProp="name">{actualPage.label}</span>
        <meta itemProp="position" content={String(mutPages.length + 1)} />
      </S.BreadcrumbItem>
    </S.Breadcrumb>
  )
})

Breadcrumb.displayName = 'Breadcrumb'
