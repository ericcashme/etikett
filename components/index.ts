import CCol from './Grid/Col'
import CRow from './Grid/Row'
import Provider, { tokens } from './theme'

export const theme = tokens
export const ComponentsProvider = Provider
export const Row = CRow
export const Col = CCol

export { Alert } from './Alert'
export { Avatar } from './Avatar'
export { Box } from './Box'
export { Breadcrumb } from './Breadcrumb'
export { Button } from './Button/Button'
export { Buttons } from './Button/Buttons'
export { CardAddress } from './CardAddress'
export { CardBlog } from './CardBlog'
export { CardProduct } from './CardProduct'
export { Checkbox } from './Checkbox'
export { Collapse } from './Collapse/Collapse'
export { CollapseItem } from './Collapse/CollapseItem'
export { CookieConsent } from './CookieConsent'
export { Icon } from './Icon'
export { NewIcon } from './Icon/NewIcon'
export { IconsList } from './Icon/IconsList'
export { Input } from './Input'
export { HorizontalScroll } from './HorizontalScroll'
export { Link } from './Link'
export { List } from './List/List'
export { ListItem } from './List/ListItem'
export { LinearGradient } from './LinearGradient'
export { Loader } from './Loader'
export { Logo } from './Logo'
export { Modal } from './Modal'
export { Placeholder } from './Placeholder'
export { RatingStar } from './Rating'
export { Select } from './Select'
export { SelectCard } from './SelectCard'
export { Tag } from './Tag'
export { Text } from './Text'
export { Textarea } from './Textarea'
export { Tooltip } from './Tooltip'
export { User } from './User'
