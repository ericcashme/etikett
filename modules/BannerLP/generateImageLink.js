import { cloudFront } from 'utils'

const generateImageLink = (path, windowWidth) => {
  if (!path || typeof path === 'number' || Array.isArray(path)) {
    return null
  }

  if (typeof windowWidth !== 'number') {
    return null
  }

  const { mobile, tablet, desktop } = path

  let url = desktop

  if (tablet && windowWidth < 992) {
    url = tablet
  }

  if (mobile && windowWidth <= 600) {
    url = mobile
  }

  if (url) {
    return `${cloudFront}${url}`
  }

  return ''
}

export default generateImageLink
