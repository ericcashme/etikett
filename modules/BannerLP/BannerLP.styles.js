import { Row, theme } from 'components'
import styled from 'styled-components'

const { breakpoint, space, color, font } = theme

export const BannerRow = styled(Row)`
  position: relative;
  margin-bottom: ${space.xl};
`

export const Imagem = styled.img`
  display: block;
  width: 100%;
  height: auto;
`

export const Content = styled.div`
  color: ${color.inverse.pure};
  position: absolute;
  left: 1.76rem;
  top: ${space.xl4};
  width: 45%;
  line-height: 16px;

  @media (min-width: ${breakpoint.xl}) {
    left: ${space.xl3};
    top: ${space.xl5};
  }

  h1 {
    font-size: ${font.size.h5};
    overflow: hidden;
    text-overflow: ellipsis;
    display: -webkit-box;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;

    @media (min-width: ${breakpoint.md}) {
      font-size: ${font.size.h3};
    }
  }

  p {
    margin-bottom: ${space.md};
  }

  div {
    display: none;

    @media (min-width: ${breakpoint.xl}) {
      display: block;
    }
  }
`
