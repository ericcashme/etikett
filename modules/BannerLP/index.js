/* eslint-disable react/no-danger */
import React, { memo, useEffect, useState } from 'react'

import { Col } from 'components'
import { useWindowSize } from 'utils'
import PropTypes from 'prop-types'

import * as S from './BannerLP.styles'
import generateImageLink from './generateImageLink'

const BannerLP = ({ content, showTitle }) => {
  const windowSize = useWindowSize()
  const { path, seo, content_title, content_text } = content
  const [imageLink, setImageLink] = useState()
  const [imageHeight, setImageHeight] = useState(337)

  useEffect(() => {
    if (windowSize) {
      setImageLink(() => generateImageLink(path, windowSize.width))
      setImageHeight((oldHeight) => {
        if ((!path.desktop || path.desktop === '') && path.mobile && path.mobile !== '') {
          return 620
        }

        return oldHeight
      })
    }
  }, [path, windowSize])

  return (
    <S.BannerRow isFull>
      <Col>
        {imageLink && <S.Imagem src={imageLink} alt={seo.alt} title={seo.title} width={1080} height={imageHeight} />}
        <S.Content>
          {showTitle && <h1>{content_title}</h1>}
          {content_text && <div dangerouslySetInnerHTML={{ __html: content_text }} />}
        </S.Content>
      </Col>
    </S.BannerRow>
  )
}

BannerLP.propTypes = {
  content: PropTypes.shape({
    path: PropTypes.shape({
      mobile: PropTypes.string,
      desktop: PropTypes.string,
      tablet: PropTypes.string,
    }),
    seo: PropTypes.shape({
      alt: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    }),
    content_title: PropTypes.string.isRequired,
    content_text: PropTypes.string.isRequired,
  }).isRequired,
  showTitle: PropTypes.bool,
}

BannerLP.defaultProps = {
  showTitle: false,
}

export default memo(BannerLP)
