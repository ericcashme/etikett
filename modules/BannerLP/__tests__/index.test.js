import React from 'react'
import { ComponentsProvider } from 'components'
import { render } from '@testing-library/react'

import BannerLP from '../BannerLP'

const bannerProps = {
  content: {
    content_text: `<p><br />
      - Conheça os produtos mais vendidos da Printi</p>
      <p>- Aproveite os acabamentos especiais de cada produto</p>
      <p>- Divulgue sua marca com produtos personalizados</p>
      <p>- Faça online de forma prática e rápida</p>
      <p></p>`,
    content_title: 'Mais Vendidos',
    path: {
      desktop: '2018-04/1524162808_1122x350px-banner-landing.jpg',
      mobile: '2018-04/1524162808_1122x350px-banner-landing.jpg',
    },
    seo: {
      alt: 'Confira os materiais mais vendidos na melhor gráfica online do Brasil 2',
      title: 'Confira os materiais mais vendidos na melhor gráfica online do Brasil 2',
    },
  },
  showTitle: true,
}

const BannerLPComponent = (props) => (
  <ComponentsProvider>
    <BannerLP {...props} />
  </ComponentsProvider>
)

describe('<BannerLP />', () => {
  it('Expect to render BannerLP with snapshot', () => {
    expect(<BannerLPComponent {...bannerProps} />).toMatchSnapshot()
  })

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(<BannerLPComponent {...bannerProps} />)
    expect(spy).not.toHaveBeenCalled()
  })
})
