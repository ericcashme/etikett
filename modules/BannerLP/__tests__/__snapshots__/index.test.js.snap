// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerLP /> Expect to render BannerLP with snapshot 1`] = `
<BannerLPComponent
  content={
    Object {
      "content_text": "<p><br />
      - Conheça os produtos mais vendidos da Printi</p>
      <p>- Aproveite os acabamentos especiais de cada produto</p>
      <p>- Divulgue sua marca com produtos personalizados</p>
      <p>- Faça online de forma prática e rápida</p>
      <p></p>",
      "content_title": "Mais Vendidos",
      "path": Object {
        "desktop": "2018-04/1524162808_1122x350px-banner-landing.jpg",
        "mobile": "2018-04/1524162808_1122x350px-banner-landing.jpg",
      },
      "seo": Object {
        "alt": "Confira os materiais mais vendidos na melhor gráfica online do Brasil 2",
        "title": "Confira os materiais mais vendidos na melhor gráfica online do Brasil 2",
      },
    }
  }
  showTitle={true}
/>
`;
